src/Adapter/CombinationDataProvider.php

```diff
@@ -180,6 +200,71 @@ public function completeCombination($attributesCombinations, $product)
         ];
     }
 
+    /**
+     * @param int $productId
+     * @param int $limit
+     * @param int $offset
+     * @param array $filters
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public function getProductCombinations(int $productId, ?int $limit = null, ?int $offset = null, array $filters = []): array
+    {
+        $qb = $this->getCombinationsQueryBuilder($productId, $filters)
+            ->select('pa.*')
+            ->setParameter('productId', $productId)
+        ;
+
+        if ($offset) {
+            $qb->setFirstResult($offset);
+        }
+
+        if ($limit) {
+            $qb->setMaxResults($limit);
+        }
```

Comment:
```suggestion
        $qb = $this->getCombinationsQueryBuilder($productId, $filters)
            ->select('pa.*')
            ->setParameter('productId', $productId)
            ->setFirstResult($offset)
            ->setMaxResults($limit)
        ;
```

Reply:
these are optional params, I cannot set null as FirstResult or MaxResults

Reply:
Default values for FirstResult & MaxResults are null so you can set to `null`.